### PR TITLE
Add imgui module

### PR DESCRIPTION
**Pull request by Meakk**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
     - name: Build and package F3D
       uses: ./source/.github/actions/f3d-superbuild/
       with:
-        f3d_version: origin/master
+        f3d_version: origin/imgui2
```
Comment (Meakk) [inline .github/workflows/ci.yml:56]: todo: revert this file

File: versions.cmake [hunk 2]
```diff
@@ -44,7 +49,7 @@ superbuild_set_selectable_source(f3d
     DOWNLOAD_NAME f3d-v2.5.0.tar.gz
     URL_MD5 f2db6b0b13b8a7b2a447a84ad2a258b1
   SELECT git CUSTOMIZABLE DEFAULT
-    GIT_REPOSITORY "https://github.com/f3d-app/f3d.git"
+    GIT_REPOSITORY "https://github.com/Meakk/f3d.git"
```
Comment (Meakk) [inline versions.cmake:52]: todo: revert this line

File: versions.cmake [hunk 3]
```diff
@@ -28,6 +28,11 @@ superbuild_set_revision(openvdb
   DOWNLOAD_NAME openvdb-v11.0.0.tar.gz
   URL_MD5 025f4fc4db58419341a4991f1a16174a)
 
+superbuild_set_revision(imgui
+  URL "https://github.com/ocornut/imgui/archive/refs/tags/v1.91.5.tar.gz"
+  DOWNLOAD_NAME openvdb-v1.91.5.tar.gz
```
Comment (Meakk) [inline versions.cmake:33]: todo: change the download name `openvdb`->`imgui`

File: projects/patches/imgui-cmake-build.patch [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
++include("${CMAKE_CURRENT_LIST_DIR}/imgui-targets.cmake")
+diff --git a/imgui.h b/imgui.h
+index d6286592..c91cb811 100644
+--- a/imgui.h
++++ b/imgui.h
+@@ -65,6 +65,8 @@ Index of this file:
+ 
+ #ifndef IMGUI_DISABLE
+ 
++#include "imgui_export.h"
```
Comment (mwestphal) [inline projects/patches/imgui-cmake-build.patch:105]: why is this needed ?

**Review by mwestphal — APPROVED** [review]
one small question though

File: projects/patches/imgui-cmake-build.patch (hunk 4 above)
Comment (Meakk) [reply projects/patches/imgui-cmake-build.patch:105]: Because imgui has an `IMGUI_API` macro that does nothing and it's left to the dev to do the `dllexport` stuff on Windows if we want to build it as a shared library.  
I used CMake to generate `IMGUI_API` macro defined in our own `imgui_export.h` file.
